pkg/datapath/linux/routing/migrate.go

```diff
@@ -0,0 +1,275 @@
+// Copyright 2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package linuxrouting
+
+import (
+	"fmt"
+
+	"github.com/cilium/cilium/pkg/datapath/linux/linux_defaults"
+	v2 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2"
+	"github.com/cilium/cilium/pkg/node"
+	"github.com/cilium/cilium/pkg/option"
+
+	"github.com/vishvananda/netlink"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// MigrateENIRoutingTable migrates the egress rules inside the Linux routing
+// policy database (rpdb) for ENI IPAM mode.
+//
+// We rely on the ability to fetch the CiliumNode resource because we need to
+// fetch the number associated with the ENI device. The CiliumNode resource
+// contains this information in the Status field.
+//
+// This function should be invoked before exposing the Cilium API because we
+// want to ensure that no workloads are scheduled while this modification is
+// taking place. This migration is related to a bug
+// (https://github.com/cilium/cilium/issues/14336) where an ENI has an ifindex
+// that equals the main routing table number (253-255), causing the rules and
+// routes to be created using the wrong table ID, which could end up
+// blackholing most traffic on the node.
+func MigrateENIRoutingTable(getter ciliumNodeGetter) []error {
+	if option.Config.IPAM != option.IPAMENI {
+		return nil
+	}
+
+	var errors []error
+
+	rules, err := netlink.RuleList(netlink.FAMILY_V4)
+	if err != nil {
+		errors = append(errors, err)
+		log.WithError(err).Error("Failed to list rules, cannot migrate ENI routing table.")
+		return errors
+	}
+
+	for _, r := range rules {
```

Comment:
I wonder, is it possible to dump all rules in priority `RulePriorityEgress` and then iterate those specifically, rather than dumping all possible rules and iterating all of them to find the ones we care about?

Reply:
Yes it is possible and I've [implemented support](https://github.com/vishvananda/netlink/pull/538) for that in netlink, but in v1.7, our netlink version does not include it. I believe we have this capability in v1.8 and onwards.